Lib/profiling/sampling/gecko_collector.py

```diff
@@ -0,0 +1,451 @@
+import json
+import os
+import platform
+import sys
+import time
+from typing import Dict, List, Optional, Tuple, Any
+
+from .collector import Collector, THREAD_STATE_RUNNING
+
+
+class GeckoCollector(Collector):
+    def __init__(self, *, skip_idle=False):
+        self.skip_idle = skip_idle
+        self.start_time = time.time() * 1000  # milliseconds since epoch
+
+        # Global string table (shared across all threads)
+        self.global_strings = ["(root)"]  # Start with root
```

Comment:
ℹ️ **Maintainability**

Using a mutable default value like a list can lead to unexpected behavior if the list is modified elsewhere. Consider using a tuple if the list is not meant to be modified.

**Suggested fix:**
```suggestion
        self.global_strings = ("(root)",)
```